### ISSUE TITLE
Toggle/Disable Scrollbars?

### ISSUE THREAD

**Issue by Celti** [flxzt/rnote#139]
>**Is your feature request related to a problem? Please describe.**

I'm using rnote on a convertible laptop with a touchscreen and stylus. Whenever I'm drawing near the right edge of the canvas, the side of my hand tends to trigger a touch on the scrollbar and sends the current view flying before hand recognition kicks in so i can properly use the stylus.

> **Describe the solution you'd like**

As you can pan around with touch controls, _and_ with the middle mouse button, the most obvious solution seems to be an option to toggle the scrollbars, alongside ‘Draw with touch input.’

>**Describe alternatives you've considered**

I considered tweaking hand recognition, but I'm not sure that would feasible within the app and have no real idea of where to start looking.

>**Additional context**

I'm willing to attempt a PR adding this, but this is a bit of an intimidating project and I figured it would be best to get some guidance first.

**Comment by flxzt** [comment]
That is a good idea. A couple of thoughts from me:

- We would need to test if it is actually the scrollbar that is responsible for moving the canvas. It could also be that some touch events are being passed through to the touch drag gesture ( see `setup_controllers()` in `appwindow/mod.rs` for registered event controllers / gestures )  
- The palm detection part is happening in libinput or GTK, I don't think either that we can tweak it ourselves. But we could try to improve on top of it:
- Do we need to make disabling the scrollbars an option? We could just change their visibility (with `PolicyType::Never`) whenever we start / stop drawing (meaning a Pen Down Event is registered). Or we could disable touch gestures entirely while drawing with a pen. I think the right place for it would be somewhere in the `process_pen_down()` function in `canvas/input.rs`. If doing that, the scrollbar policy type would need to be set (with [set_policy()](https://gtk-rs.org/gtk4-rs/stable/0.4/docs/gtk4/struct.ScrolledWindow.html#method.set_policy).
- Another, more pragmatic solution would be ( if it actually is the scrollbars that are the issue): we could just swap the sides of the vertical scrollbar. It would reverse this [PR](https://github.com/flxzt/rnote/pull/100), but I think your reasons would outweigh the reasons of that PR.

**Comment by Celti** [comment]
> * We would need to test if it is actually the scrollbar that is responsible for moving the canvas. It could also be that some touch events are being passed through to the touch drag gesture ( see `setup_controllers()` in `appwindow/mod.rs` for registered event controllers / gestures )

Fair! I was assuming it was the scrollbar because it was only happening at that edge of the window, but I'll do a bit of tracing to make certain.

> * The palm detection part is happening in libinput or GTK, I don't think either that we can tweak it ourselves. But we could try to improve on top of it:

Yeah, I thought that was the case.

> * Do we need to make disabling the scrollbars an option? We could just change their visibility (with `PolicyType::Never`) whenever we start / stop drawing (meaning a Pen Down Event is registered). Or we could disable touch gestures entirely while drawing with a pen. I think the right place for it would be somewhere in the `process_pen_down()` function in `canvas/input.rs`. If doing that, the scrollbar policy type would need to be set (with [set_policy()](https://gtk-rs.org/gtk4-rs/stable/0.4/docs/gtk4/struct.ScrolledWindow.html#method.set_policy).

The first problem I see here is that I'm not sure whether the touch on the scrollbar would be registered before the pen event — what it boils down to is that I have bad drawing habits and rest my hand on the screen itself. 😛 Triggering on pen proximity might work, but it looks like libinput's `TABLET_TOOL_PROXIMITY` doesn't translate straight to a `PenEvent`?

> * Another, more pragmatic solution would be ( if it actually is the scrollbars that are the issue): we could just swap the sides of the vertical scrollbar. It would reverse this [PR](https://github.com/flxzt/rnote/pull/100), but I think your reasons would outweigh the reasons of that PR.

Huh, I started using rnote recently enough I hadn't realised that PR existed! I was happy to have the sidebar on the left to avoid having this same thing with the buttons

I'll do some testing in the morning and update this then.

**Comment by flxzt** [comment]
Did you manage to trace if the scrollbars are the cause?
I refactored some things in the `compose` branch which will be merged into main at some point, and there PenEvent does accomodate for proximity events. But
I believe the transitions into and out of the down state is where the visibility should be toggled (else it would prevent stylus users from using the scrollbar entirely.

I also already made a change which does exactly that :P sorry I you wanted to do it, but I thought since I am refactoring so much in that branch it would be annoying to dig and understand the code while it changes underneath.

Maybe you could test it out (ignore the other issues / bugs the branch currently still has) and tell if that actually fixes it?

**Comment by flxzt** [comment]
Did this resolve your issues?

**Comment by Danacus** [comment]
I've experienced the same issue too. I think that hiding scrollbars when the pen is down won't do anything meaningful, because the touchscreen usually gets disable whenever the pen is in proximity. 

My guess is that this issue occurs when accidentally touching the scrollbar with your palm while the pen is not yet near the screen.

I'm pretty sure this is related to the scrollbars, because I had the same problem in Xournal++ before I disabled the scrollbars.
